src/networking/address.rs

```diff
@@ -0,0 +1,45 @@
+
+/// IPV6 address
+pub struct IPV6Addr {
+  address: String,
+  port: u16,
+
+}
+
+/// IPV4 address
+pub struct IPV4Addr {
```

Comment:
Just a thought: If the only reason for having different structs for v4 and v6 is to differentiate them isn't it maybe better (in the sense of less code duplication) to have one generic IP struct which has address,port and kind, where kind is of an enum type which contains v4 and v6. This way you can still differentiate between them easily and it would lessen the duplication of code. Of course if the structs will differentiate further later this makes less sense.

Reply:
hm, you are right about it now. I was thinking they might differentiate in the future though. Not their fields but their functions will.

Reply:
I'm not proficient enough with rust to know if this is easily possible - having the same data structs but different functions on them. Probably it's possible by defining functions on an enum.

Reply:
For example: what if you want to convert an ipv4 address to an ipv6 address. A method for this should only be defined on the ipv4 struct.